fe/lib/components/Questionnaire/QuestionnaireBuilder/QuestionnaireBuilder.tsx

```diff
@@ -17,22 +17,33 @@ import {
   QuestionnaireCreateInput,
   convertComponentsToMutationVariables,
 } from '../components/GraphqlQueryRenderer/GraphqlQueryRenderer';
-import { FormComponent } from '../questionTypes';
+import { mapMutationVariableToFormComponent } from '../mapping';
+import { FormComponent, MutationVariableInput } from '../questionTypes';
 
 import { FormBuilder } from './FormBuilder/FormBuilder';
 
 interface QuestionnaireBuilderProps {
   hirerId?: string;
+  graphqlInputVariables?: MutationVariableInput;
   onChange?: (mutationVariables: QuestionnaireCreateInput) => void;
 }
 
 export const QuestionnaireBuilder = ({
   hirerId: externalHirerId,
+  graphqlInputVariables,
   onChange,
 }: QuestionnaireBuilderProps) => {
   const [formBuilderState, setFormBuilderState] = useState<FormComponent[]>([]);
   const [hirerId, setHirerId] = useState(externalHirerId ?? '');
 
+  useEffect(() => {
+    setFormBuilderState(
+      graphqlInputVariables
+        ? mapMutationVariableToFormComponent(graphqlInputVariables)
+        : [],
+    );
```

Comment:
Should we make the effect conditional like:

```suggestion
    if (graphqlInputVariables) {
      setFormBuilderState(
        mapMutationVariableToFormComponent(graphqlInputVariables),
      );
    }
```

An undefined input doesn't seem valid so I'm not sure we should clear our state based on it.

Reply:
Oh yep that makes sense